src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -477,6 +478,21 @@ public final class Guardrails implements GuardrailsMBean
                      format("The modification to table %s has a timestamp %s before the minimum allowable %s threshold %s",
                             what, value, isWarning ? "warning" : "failure", threshold));
 
+    public static final MaxThreshold maximumReferencedIndexesWithNonParitionRestrictedQuery =
```

Comment:
I chose to go with guardrail approach even we are not checking guardrail as such (will be visible further in down in the patch). This has a nice property of being update-able via JMX in runtime.

Reply:
Yeah, the standardization there is nice.